tests/components/smartthings/test_lock.py

```diff
@@ -0,0 +1,125 @@
+"""
+Test for the SmartThings lock platform.
+
+The only mocking required is of the underlying SmartThings API object so
+real HTTP calls are not initiated during testing.
+"""
+from pysmartthings import Attribute, Capability
+
+from homeassistant.components.smartthings import DeviceBroker, lock
+from homeassistant.components.smartthings.const import (
+    DATA_BROKERS, DOMAIN, SIGNAL_SMARTTHINGS_UPDATE)
+from homeassistant.config_entries import (
+    CONN_CLASS_CLOUD_PUSH, SOURCE_USER, ConfigEntry)
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+
+
+async def _setup_platform(hass, *devices):
```

Comment:
This has been generalized and moved to `conftest` -- see the sensor platform for usage.  No need to define this now in each test module.

Reply:
It looks like `lock` doesn't support `async_unload_entry` so I've removed that test.

In the process I noticed `test_switch.py` has a faulty test:
```
async def test_unload_config_entry(hass, device_factory):
    """Test the switch is removed when the config entry is unloaded."""
    # Arrange
    device = device_factory('Switch', [Capability.switch],
                            {Attribute.switch: 'on'})
    config_entry = await _setup_platform(hass, device)
    # Act
    await hass.config_entries.async_forward_entry_unload(
        config_entry, 'switch')
    # Assert
    assert not hass.states.get('switch.switch_1')
```

`device = device_factory('Switch'` should be `device = device_factory('Switch 1'`

Reply:
Good catch.  Please fix and add to this PR